include/openmc/tallies/filter_collision.h

```diff
@@ -0,0 +1,51 @@
+#ifndef OPENMC_TALLIES_FILTER_COLLISIONS_H
```

Comment:
You can replace the ifndefs here at the top of the file (and correspondingly remove the macro closing statement at the bottom of the file) and instead put `#pragma once` at the top of the file to achieve the same effect. Anyone who thinks we shouldn't do this can duke it out with me. There has been pushback against using this in the past out of fear of compiler compatibility, but we're totally inconsistent on that viewpoint because we use gsl, which uses `pragma once`.

Reply:
For now please keep the ifdef as is. If we decide to go to `pragma once`, let's do it all at once rather than piecemeal.